### PR TITLE
adjust docker image and add deployed, version json

### PR DESCRIPTION
**Pull request by sidler-mozilla**
- created refractr/image/
- moved refractr/refractr.sh -> refractr/image/refractr.sh
- added cfg.py to handle config values and allow env var passthrus
- added deployed and version tasks to create json files in image/
- Dockerfile to COPY image/ -> /refractr/; nginx/ -> /etc/nginx/
- updated Dockerfile ENTRYPOINT /refractr/refractr.sh
- moved logic into SHOULD_PUBLISH property
- simplified refractr image name in docker-compose
- publish now creates a tag for the ecr repository name
- .gitignore for generated files: deployed, version, ingress.yaml

### CODE REVIEW

**Comment by ziegeer** [comment]
It'd be nice to split this up into multiple PRs in the future, with more extensive descriptions "created refractr/image/" means nothing to me for example.

File: docker-compose.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: "3.7"
 
 services:
   refractr:
-    image: ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com/refractr:${REFRACTR_VERSION}
+    image: refractr:${REFRACTR_VERSION}
```
Comment (ziegeer) [inline docker-compose.yml:5]: I'm unclear how this resolves to ECR and not Dockerhub.

File: dodo.py [hunk 2]
```diff
@@ -5,92 +5,45 @@
 import json
 
 from doit.tools import LongRunning
-from subprocess import check_output, CalledProcessError, PIPE
+from functools import lru_cache
```
Comment (ziegeer) [inline dodo.py:8]: What is the lru_cache property for?  Isn't there a simpler way to do whatever it is?

File: docker-compose.yml (hunk 1 above)
Comment (sidler-mozilla) [reply docker-compose.yml:5]: https://github.com/mozilla-it/refractr/pull/46/files#diff-5ddda38bd048b0606552d6c1890f9259L269-R254

File: dodo.py (hunk 2 above)
Comment (sidler-mozilla) [reply dodo.py:8]: It caches the result of those properties, so after it is calculated once, it is never calculated again.  There is no cheaper way to provide this functionality.  Also the docs: https://docs.python.org/3/library/functools.html

File: docker-compose.yml (hunk 1 above)
Comment (sidler-mozilla) [reply docker-compose.yml:5]: `refractr` locally, tagged with ECR url only for publish.  Makes your local docker images not so damn spammy with the gigantic ECR url.

**Comment by sidler-mozilla** [comment]
> It'd be nice to split this up into multiple PRs in the future, with more extensive descriptions "created refractr/image/" means nothing to me for example.

It means I created the folder/ and started populating it with stuff.  It makes sense if you checkout the branch and look around.

**Pull request merged**